buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/shared-configuration.kt

```diff
@@ -106,11 +107,20 @@ fun Project.integrationTestUsesSampleDir(vararg sampleDirs: String) {
 }
 
 
+/**
+ * kotlin-dsl-plugins integration test depends on jars published to local repository
+ */
 fun Project.integrationTestUsesKotlinDslPlugins() {
     tasks.withType<IntegrationTest>().configureEach {
-        inputs.dir(project(":kotlinDslPlugins").buildDir.resolve("repository"))
-            .withPropertyName("localRepository")
+        inputs.files(callable {
+            project(":kotlinDslPlugins")
+                .fileTree("build/repository")
+                .matching { include("**/*.jar") }
+                .files
+                .toSortedSet()
+        }).withPropertyName("localKotlinDslPluginJars")
             .withPathSensitivity(PathSensitivity.RELATIVE)
+            .withNormalizer(ClasspathNormalizer::class)
     }
 }
 
```

Comment:
Can we also get rid of `integrationTestUsesToolingApiFatJar()` below and use "libsRepository.required = true" as well instead (in the _plugin-development_ project, which seems to be the only place where that is used)